app/controllers/application_controller.rb

```diff
@@ -1140,12 +1143,34 @@ def listicon_image(view, id = nil)
              klass.find(id)    # Read the record from the db
            end
 
-    p  = "/images/icons/"
-    pn = "#{p}new/"
+    [id, item]
+  end
+  private :listicon_get_id_item
+
+  # Return the icon classname for the list view icon of a db,id pair
+  # this always supersedes listicon_image if not nil
+  def listicon_icon(view, id = nil)
+    id, item = listicon_get_id_item(view, id)
+
+    case item
+    when ExtManagementSystem
+      item.fonticon
+    else
+      nil
+    end
+  end
+
+  # Return the image name for the list view icon of a db,id pair
+  def listicon_image(view, id = nil)
+    id, item = listicon_get_id_item(view, id)
+
+    pn = "icons/new/"
+    po = "icons/"
+    pi = "/pictures/"
```

Comment:
`pn` was there and `po` was called `p` .. I just wanted it to be obvious that we're touching these 3 places and no others. Maybe making the var names a bit more ..explicit might help? :)